docs/source/user/5minguide.rst

```diff
@@ -0,0 +1,247 @@
+`Anaconda Python distribution <https://www.anaconda.com/>`_::
+
+  $ conda install numba
+
+Numba also has wheels available::
+
+  $ pip install numba
+
+Numba can also be compiled from source (`instructions <LINK>`_)
+
+Numba is often used as a core package so its dependencies are kept to an
+absolute minimum, however, extra packages can be installed as follows to provide
+additional functionality:
+
+* ``scipy`` - enables support for compiling ``numpy.linalg`` functions
+* ``colorama`` - enables support for color highlighting in backtraces/error
+  messages
+* ``pyyaml`` - enables configuration of Numba via a YAML config file.
+* ``icc_rt`` - allows the use of the Intel SVML (high performance short vector
+  math library, x86_64 only).
+
+Will Numba work for my code?
+----------------------------
+This depends on what your code "looks like", if your code is numerically
+orientated (does a lot of math), uses NumPy a lot and/or has a lot of loops,
+then Numba is often a good choice. In these examples we'll apply the most
+fundamental of Numba's JIT decorators, ``@jit``, to try and speed up some
+functions to demonstrate what works well and what does not.
+
+Numba works well on code that "looks like" this::
+
+    from numba import jit
+    import numpy as np
+
+    x = np.arange(100).reshape(10, 10)
+
+    @jit
+    def go_fast(a): # Function is compiled and runs in machine code
+        trace = 0
+        for i in range(a.shape[0]):   # Numba likes loops
+            trace += np.tanh(a[i, i]) # Numba likes NumPy functions
+        return a + trace              # Numba likes NumPy broadcasting
+
+    print(go_fast(x))
+
+
+It works reasonably well on code with well defined numerically orientated loops
+(it will "lift" out the loops and compile those to machine code and run the rest
+in the Python interpreter)::
+
+    from numba import jit
+    import numpy as np
+
+    x = np.arange(20)
+
+    @jit
+    def loop_lift(x):
+        a = object() # Numba runs this in the interpreter
+        acc = 0
+        for i in x:  # Numba will compile this loop to machine code!
+            acc +=i
+        return a, acc
+
+    print(loop_lift(x))
+
+
+It won't work very well, if at all, on code that "looks like" this::
+
+    from numba import jit
+    import pandas as pd
+
+    x = {'a': [1, 2, 3], 'b': [20, 30, 40]}
+
+    @jit
+    def use_pandas(a): # Function will not benefit from Numba jit
+        df = pd.DataFrame.from_dict(a) # Numba doesn't know about pd.DataFrame
+        df += 1                        # Numba doesn't understand what this is
+        return df.cov()                # or this!
+
+    print(use_pandas(x))
+
+Note that Pandas is not understood by Numba and as a result Numba would simply
+run this code via the interpreter but with the added cost of the Numba internal
+overheads!
+
+The ``nopython`` mode *gotcha*.
+-------------------------------
+Pretty much all first time users are caught by this so it deserves a mention...
```

Comment:
I would shorten this section and write it assuming nopython mode is the standard way to use `@jit`, mention briefly that Numba has limited support for mixed mode via loop lifting and maybe link to somewhere in the docs that give more information.